### DISCUSSION TITLE
"Previous value" always indentical valua as the actual measured value

### DISCUSSION THREAD

**Discussion by topsurfer** [jomjol/AI-on-the-edge-device#2534] [Q&A]
Hi,
I try to understand the field (value) from the field "Previous value" in the Web-Gui (Overview):
In my understanding after a new run the last value will be the "previous value". And the measured value is the "Value".

My test run is every 3 minutes, and always there is a change from the values.
BUT after every run, the fields "Value" **and** Previous value" are having the new (actual) value. 

Is there a bug or has I a misunderstanding about "Previous value" in the Web-GUI ?
( At: "Settings =>Configuration => Postprocessing => Previous value" there is the default activ "true" )

**Only a Example:**
Run 0:
Value: 100
Previous value: nA (?)

Run 1:
Value: 120
Previous value: 120

Run 2:
Value: 134
Previous value: 134
...
...

Real Example /Screens:
19:31:
https://abload.de/img/19-312rit7.jpg

19:34:
https://abload.de/img/19-348cesa.jpg

**Comment by caco3** [comment]
The `PreValue` gets updated with the `Value` after every successful run.

This might look odd on the UI, but this new PreValue gets used in the next round, this needs to be up-to-date.

**Comment by signrossi** [comment]
i also see this on MQTT Messages

i can find that prevalue is updated in ClassFlowPostProcessing::SetPreValue, but cannot detect if this is done before or after the MQTT Message goes out
